backend/siarnaq/api/episodes/challonge.py

```diff
@@ -68,31 +74,149 @@ def bulk_add_participants(tournament_url, participants):
+                # matches could be open _or done_.
+                # !!! This is also _really hard_ right now
+                # cuz it involves match deletion which is really hard.
+                # Track in #549
+                if state != "open":
+                    # For later, have this raise a more specific exception.
+                    # Then have the caller handle this return
+                    # and translate it into an HTTP response.
+                    raise RuntimeError(
+                        "The bracket service's round does not only\
+                            have matches that are ready."
+                    )
+                challonge_matches.append(item)
+
+            case {"type": "participant", "id": id}:
+                challonge_participants[id] = item
+
+    match_objects = []
+    match_participant_objects = []
+
+    for challonge_match in challonge_matches:
+        match_object = apps.get_model("compete", "Match")(
+            episode=round.tournament.episode,
+            tournament_round=round,
+            alternate_order=True,
+            is_ranked=False,
+            challonge_id=challonge_match["id"],
+        )
+        match_objects.append(match_object)
+
+        # Note that Challonge 1-indexes its player indexes
+        # while Saturn 0-indexes.
+        for (challonge_player_index, saturn_player_index) in (
+            ("player1", 0),
+            ("player2", 1),
+        ):
+            # This looks ugly but it's how to parse through the Challonge-related data.
+            challonge_participant_id = challonge_match["relationships"][
+                challonge_player_index
+            ]["data"]["id"]
+            challonge_participant_misc_key = challonge_participants[
+                challonge_participant_id
+            ]["attributes"]["misc"]
+            team_id, submission_id = (
+                int(_) for _ in challonge_participant_misc_key.split(",")
+            )
+
+            match_participant_object = apps.get_model("compete", "MatchParticipant")(
+                team_id=team_id,
+                submission_id=submission_id,
+                match=match_object,
+                player_index=saturn_player_index,
+                challonge_id=challonge_participant_id,
+            )
+            match_participant_objects.append(match_participant_object)
+
+    return match_objects, match_participant_objects
+
+
+def update_match(tournament_id, match_id, match):
+    url = f"{URL_BASE}tournaments/{tournament_id}/matches/{match_id}.json"
+
+    # Wrangle the Match object into a format Challonge likes.
+    # In particular, Challonge wants an array of objects,
+    # each of which represents a participant's data.
+    # The data is id, score, and whether or not they advance.
+
+    participants_for_challonge = []
+
+    # Build this array while also tracking high score (for computing who advances).
+    high_score = -1
```

Comment:
Can `high_score` just be `high_score = max(participant.score for participant in ...)`